src/crimson/os/seastore/onode_manager/staged-fltree/node.cc

```diff
@@ -311,12 +311,18 @@ eagain_future<Node::search_result_t> Node::lower_bound(
 }
 
 eagain_future<std::pair<Ref<tree_cursor_t>, bool>> Node::insert(
-    context_t c, const key_hobj_t& key, value_config_t vconf)
+    context_t c,
+    const key_hobj_t& key,
+    value_config_t vconf,
+    Ref<Node>&& this_ref)
 {
   return seastar::do_with(
-    MatchHistory(), [this, c, &key, vconf](auto& history) {
+    MatchHistory(), [this, c, &key, vconf,
+                     this_ref = std::move(this_ref)] (auto& history) mutable {
       return lower_bound_tracked(c, key, history
-      ).safe_then([c, &key, vconf, &history](auto result) {
+      ).safe_then([c, &key, vconf, &history,
+                   this_ref = std::move(this_ref)] (auto result) mutable {
+        this_ref.reset();
```

Comment:
this is confusing, as `this_ref` should be destructed along with this closure.

Reply:
added comment to explain.